### DISCUSSION TITLE
Can't get it to work with smtp2go.com

### DISCUSSION THREAD

**Discussion by spherman** [deflomu/Roundcube-SMTP-per-Identity-Plugin#37] [General]
My server hosts many domains.  We currently use mailchannels for our outbound traffic.

I have a few clients who are complaining that their messages are not being received so I signed up for SMTP2go.com.  It works great with Outlook, but of course... they use RoundCube.

I found your plugin and was excited that it might be a solution to our problem.

I downloaded it and put it in place.  The settings showed up and I populated the SMTP2go.com settings.

tls://mail.smtp2go.com:2525
username
password

Unfortunately, RC sent out through mailchannels despite those settings.  Yes.  I was using the identity that had the SMTP settings configured.

I added an exclude file for mail channels thinking that it might be overriding the settings I put in place, but then it just sent from the IP address of the server instead.  That won't work for us because our ISP is on a couple pesky RBL lists.  

I am out of ideas.  Do you have any ideas?  Am I missing something?

**Comment by jthouse27** [comment]
I'm having the same issue. This would be so powerful if I can get it to work with SMTP2GO. Did you ever find a solution?

**Comment by spherman** [comment]
Sadly no.

 

I just made my client use Outlook.

 

Scott

**Comment by jthouse27** [comment]
I spent the whole day trying to get it to work. Finally gave up on editing the plugin and used Exim Configuration Manager edits to connect my client's Roundcube to SMTP2GO. It's not a scalable solution but works for this single user's account. I love how this plugin works and I hope the developers can help us get this solved.

**Comment by spherman** [comment]
Interesting.  Good for you.

 

You should post instructions here in case anyone else come across this issue.

 

Scott

**Comment by jthouse27** [comment]
WHM > Exim Configuration Manager > Advanced Editor

### Section: AUTH
```
# Authenticate to SMTP2GO for <clientdomain.com>
smtp2go_<clientName>:
  driver = plaintext
  public_name = LOGIN
  client_send = ":<clientSmtpUsername>:<clientSmtpPassword>"
```

### Section: ROUTERSTART
```
# Route mail from *@<clientdomain.com> via SMTP2GO
smtp2go_for_<clientName>:
  driver = manualroute
  senders = *@<clientdomain.com>
  transport = smtp2go_smtp
  route_list = * smtp.smtp2go.com::587 byname
  host_find_failed = defer
```

### Section: TRANSPORTSTART
```
# TLS‐enforcing SMTP transport for SMTP2GO
smtp2go_smtp:
  driver = smtp
  port = 587
  hosts_require_tls = *
  hosts_require_auth = mail.smtp2go.com
```

Don't include '<' or '>' signs in the actual code; those indicate placeholders. You should be able to change port numbers without any issues, but I haven't tested that.

**Comment by deflomu** [comment]
Hey everyone sorry for the late reply. I don't use roundcube for a very long time so I cannot test the plug-in or develop any new features. Sorry for that

**Comment by spherman** [comment]
Great!  What goes in this setting?

 

smtp2go_for_<clientName>:

 

Also, are the “clientSmtpUsername” and “clientSmtpPassword” are the SMTP2go.com username and password?

 

Scott

**Comment by jthouse27** [comment]
> smtp2go_for_<clientName>:

This is just a cute little nickname you give to the function (Idk if it should be called a "function"). It's just a label and I put my clients cPanel username in there to keep things clean. If I ever need to do this for another client, I'll create another one with the next client's cPanel username.

> Also, are the “clientSmtpUsername” and “clientSmtpPassword” are the SMTP2go.com username and password?

Yes, that's the SMTP User username and password, not necessarily for the SMTP2GO login.

Hope that helps!

**Comment by spherman** [comment]
Thanks!
